joomla/joomla-cms
# Implement featured articles filter for article category/blog menu types.

Making this PR for @Klipper :)

In the module Articles category it is possible to select how featured articles are shown:
![module_articles_category](https://cloud.githubusercontent.com/assets/359377/5686257/bc6c33a2-9841-11e4-8e45-408156184f85.png)

This same filter is not available when making a menu item for a category list or category blog layout of articles. This PR implements this feature.

**Test instructions**
 After applying the PR a new option will show up in 2 places:
- Global configuration of articles under **_Shared options**_
  ![featured_articles_global_config](https://cloud.githubusercontent.com/assets/359377/5686314/20d01fde-9842-11e4-90a1-b69b9d5b42fd.png)
- Editing a menu item of the type **_Category Blog**_ and **_Category List**_ under **_List Layouts**_
  ![featured_articles_menu_config](https://cloud.githubusercontent.com/assets/359377/5686365/74645e8a-9842-11e4-9a7e-b2909390cfef.png)

This option works the same as all other options. The global article setting is used unless the setting is changed in the menu item as this will override the global setting.

There are 3 settings:
- Show
  This will show regular and featured articles
- Hide
  This will hide the featured articles
- Only
  This will show only the featured articles

Create a menu item of both the Category Blog and Category List type and test to see on the site if the correct articles are shown based on the settings chosen.

**Backwards compatibility**
As this is a new feature there are no backwards compatibility issues to expect. The option simply leverages a setting already available in a core module.
